pkg/generate/app/sourcelookup.go

```diff
@@ -597,7 +597,11 @@ func CloneAndCheckoutSources(repo git.Repository, remote, ref, localDir, context
 		}
 	}
 	if len(contextDir) > 0 {
-		glog.V(5).Infof("Using context directory %q. The full source path is %q", contextDir, filepath.Join(localDir, contextDir))
+		if _, err := os.Stat(filepath.Join(localDir, contextDir)); os.IsNotExist(err) {
+			return "", fmt.Errorf("supplied context directory '%s' does not exist in '%s' repository", contextDir, remote)
+		} else {
+			glog.V(5).Infof("Using context directory %q. The full source path is %q", contextDir, filepath.Join(localDir, contextDir))
+		}
```

Comment:
looks like there are other paths where a contextdir is consumed that this won't catch, e.g.:
https://github.com/openshift/origin/blob/master/pkg/generate/app/sourcelookup.go#L249

i think maybe you need to be checking the path later in the code flow.